packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -1706,6 +1709,61 @@ function rerenderOpaqueIdentifier(): OpaqueIDType | void {
   return id;
 }
 
+function mountRefresh() {
+  const cacheInstance: CacheInstance = readContext(CacheContext);
+  return mountCallback(refreshCache.bind(null, cacheInstance), [cacheInstance]);
+}
+
+function updateRefresh() {
+  const cacheInstance: CacheInstance = readContext(CacheContext);
+  return updateCallback(refreshCache.bind(null, cacheInstance), [
+    cacheInstance,
+  ]);
+}
+
+function refreshCache<T>(
+  cacheInstance: CacheInstance,
+  seedKey: ?() => T,
+  seedValue: T,
+) {
+  const provider = cacheInstance.provider;
+
+  // Inlined startTransition
+  // TODO: Maybe we shouldn't automatically give this transition priority. Are
+  // there valid use cases for a high-pri refresh? Like if the content is
+  // super stale and you want to immediately hide it.
```

Comment:
There are many cases where it has to go in the same update as other things because otherwise you end up showing stale data while you wait. At the very least it probably shouldn't automatically be bumped out of that update.

It could maybe be a warning but seems plausible that you might need this for discrete input that immediately invalidates something.